src/hooks/useTransformRoutesRedirectProperties.js

```diff
@@ -0,0 +1,61 @@
+import PATHS from 'pages/paths'
+import firebase from 'firebase/compat/app'
+import { getAuth } from 'firebase/auth'
+import { useAuthState } from 'react-firebase-hooks/auth'
+import { useLocation } from 'react-router-dom'
+import { useMemo } from 'react'
+import { useUser } from 'domains/User/context'
+
+const { VIPPS_LOGIN_CALLBACK } = PATHS.UNAUTHENTICATED
+const PASSWORD_AUTH_PROVIDER = 'password'
+
+const useTransformRoutesRedirectProperties = () => {
+  const location = useLocation()
+  const [userAuth, authLoading, authError] = useAuthState(firebase.auth())
+  const { user, loading } = useUser() // uncomment this part if you need to add extra conditions based on data from the DB
+
+  // Merge all loadings to one loading
+  // Artificial loading required for the case when user is logged in but upload data from DB not started yet
+  // to prevent rewriting user data
+  const combinedLoading = useMemo(
```

Comment:
Absolutely unnecessary to use useMemo here, there are no difficult computations at all...